docs/_docs/2_2_writing_data.md

```diff
@@ -176,15 +176,49 @@ In some cases, you may want to migrate your existing table into Hudi beforehand.
 
 ## Datasource Writer
 
-The `hudi-spark` module offers the DataSource API to write (and also read) any data frame into a Hudi table.
-Following is how we can upsert a dataframe, while specifying the field names that need to be used
-for `recordKey => _row_key`, `partitionPath => partition` and `precombineKey => timestamp`
+The `hudi-spark` module offers the DataSource API to write (and read) a Spark DataFrame into a Hudi table. There are a number of options available:
 
+**`HoodieWriteConfig`**:
+
+**TABLE_NAME** (Required)<br>
+
+
+**`DataSourceWriteOptions`**:
+
+**RECORDKEY_FIELD_OPT_KEY** (Required): Primary key field(s). Nested fields can be specified using the dot notation eg: `a.b.c`. When using multiple columns as primary key use comma separated notation, eg: `"col1,col2,col3,etc"`. Single or multiple columns as primary key specified by `KEYGENERATOR_CLASS_OPT_KEY` property.<br>
+Default value: `"uuid"`<br>
+
+**PARTITIONPATH_FIELD_OPT_KEY** (Required): Columns to be used for partitioning the table. To prevent partitioning, provide empty string as value eg: `""`. Specify partitioning/no partitioning using `KEYGENERATOR_CLASS_OPT_KEY` and if using hive `HIVE_PARTITION_EXTRACTOR_CLASS_OPT_KEY`<br>
```

Comment:
hi, again the `HIVE_PARTITION_EXTRACTOR_CLASS_OPT_KEY` is not needed when not syncing to hive.

Reply:
This notes that HIVE_PARTITION_EXTRACTOR_CLASS_OPT_KEY only needs to be specified if using hive. Do you have a suggestion on how to phrase it differently?

Reply:
how about 
Specify partitioning/no partitioning using `KEYGENERATOR_CLASS_OPT_KEY` and using `HIVE_PARTITION_EXTRACTOR_CLASS_OPT_KEY` when syncing to hive ?

Reply:
Building on that, how about:?
Specify partitioning/no partitioning using `KEYGENERATOR_CLASS_OPT_KEY`. If synchronizing to hive, also specify using `HIVE_PARTITION_EXTRACTOR_CLASS_OPT_KEY`.

Reply:
> Building on that, how about:?
> Specify partitioning/no partitioning using `KEYGENERATOR_CLASS_OPT_KEY`. If synchronizing to hive, also specify using `HIVE_PARTITION_EXTRACTOR_CLASS_OPT_KEY`.

resonable.